OneSignalSDK/onesignal/src/main/java/com/onesignal/OSNotificationWorkManager.java

```diff
@@ -33,6 +45,19 @@ static void beginEnqueueingWork(Context context, String osNotificationId, int an
                 .setInputData(inputData)
                 .build();
 
+        // Duplicate control
+        // Keep in memory in going processed notifications, to avoid fast duplicates that already finished work process but are not completed yet
+        // enqueueUniqueWork might not be enough, if the work already finished then the duplicate notification work might be queued again
+        if (!Strings.isEmptyOrWhitespace(osNotificationId)) {
+            if (notificationIds.contains(osNotificationId)) {
```

Comment:
Can we combine this logic with `OneSignal.isDuplicateNotification` somehow? It isn't obvious that we have two ways of checking for duplicates.